### PR TITLE
[HDInsight] Add two new APIs: listhosts and restart node

### PR DESCRIPTION
**Pull request by aim-for-better**
Add two new APIs: list hosts and restart node

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by raych1** [comment]
@aim-for-better , here's the detail log for model validation failures. In the example file of `RestartVirtualMachinesOperation.json`, the parameter of `hosts` should be array of string instead of array of object.

```
inner:
             [ { code: 'INVALID_TYPE',
                 params: [ 'string', 'object' ],
                 message: 'Expected type string but found type object',
                 path: [ 1 ],
                 title:
                  '{"path":["definitions","RestartHostsParameters","items"],"position":{"line":158,"column":16},"url":"D:\\\\git\\\\github\\\\azure-rest-api-specs\\\\specification\\\\hdinsight\\\\resource-manager\\\\Microsoft.HDInsight\\\\stable\\\\2018-06-01-preview\\\\virtualMachines.json","directives":{}}' },
               { code: 'INVALID_TYPE',
                 params: [ 'null', 'object' ],
                 message: 'Expected type null but found type object',
                 path: [ 1 ],
                 title:
                  '{"path":["definitions","RestartHostsParameters","items"],"position":{"line":158,"column":16},"url":"D:\\\\git\\\\github\\\\azure-rest-api-specs\\\\specification\\\\hdinsight\\\\resource-manager\\\\Microsoft.HDInsight\\\\stable\\\\2018-06-01-preview\\\\virtualMachines.json","directives":{}}' } ] }

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/virtualMachines.json [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HDInsight/clusters/{clusterName}/listHosts": {
```
Comment (pilor) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/virtualMachines.json]: >listHosts [](start = 125, length = 9)

this should be a POST call

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/virtualMachines.json [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        }
+      },
+      "readOnly": true
+    },
+    "RestartHostsParameters": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      },
+      "description": "The list of roles where script will be executed."
```
Comment (pilor) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/virtualMachines.json]: >The list of roles where script will be executed [](start = 22, length = 47)

is this description accurate?

**Review by pilor — CHANGES_REQUESTED** [review]
:clock1:

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/virtualMachines.json (hunk 2 above)
Comment (aim-for-better) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/virtualMachines.json]: > > The list of roles where script will be executed [](start = 22, length = 47)
> 
> is this description accurate?

Updated

Comment (aim-for-better) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/virtualMachines.json]: > > The list of roles where script will be executed [](start = 22, length = 47)
> 
> is this description accurate?

Updated

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/virtualMachines.json (hunk 1 above)
Comment (aim-for-better) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/virtualMachines.json]: > > listHosts [](start = 125, length = 9)
> 
> this should be a POST call

waiting me to check with backend team

Comment (aim-for-better) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/virtualMachines.json]: > > listHosts [](start = 125, length = 9)
> 
> this should be a POST call

Updated to post

**Review by pilor — APPROVED** [review]

**Comment by aim-for-better** [comment]
> RestartHostsP

fixed.

> @aim-for-better , here's the detail log for model validation failures. In the example file of `RestartVirtualMachinesOperation.json`, the parameter of `hosts` should be array of string instead of array of object.
> 
> ```
> inner:
>              [ { code: 'INVALID_TYPE',
>                  params: [ 'string', 'object' ],
>                  message: 'Expected type string but found type object',
>                  path: [ 1 ],
>                  title:
>                   '{"path":["definitions","RestartHostsParameters","items"],"position":{"line":158,"column":16},"url":"D:\\\\git\\\\github\\\\azure-rest-api-specs\\\\specification\\\\hdinsight\\\\resource-manager\\\\Microsoft.HDInsight\\\\stable\\\\2018-06-01-preview\\\\virtualMachines.json","directives":{}}' },
>                { code: 'INVALID_TYPE',
>                  params: [ 'null', 'object' ],
>                  message: 'Expected type null but found type object',
>                  path: [ 1 ],
>                  title:
>                   '{"path":["definitions","RestartHostsParameters","items"],"position":{"line":158,"column":16},"url":"D:\\\\git\\\\github\\\\azure-rest-api-specs\\\\specification\\\\hdinsight\\\\resource-manager\\\\Microsoft.HDInsight\\\\stable\\\\2018-06-01-preview\\\\virtualMachines.json","directives":{}}' } ] }
> ```

fixed. Now all ci checks have passed

**Pull request merged**